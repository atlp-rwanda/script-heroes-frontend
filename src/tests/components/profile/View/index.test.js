import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { useDispatch, useSelector } from 'react-redux';

import ViewProfile from '../../../../components/profile/View';

const mockDispatch = jest.fn();
const mockSelector = jest.fn();

jest.mock('react-redux', () => ({
  useSelector: () => mockSelector,
  useDispatch: () => mockDispatch,
}));

describe('<ViewProfile/>', () => {
  let wrapper;
  beforeEach(() => {
    act(() => {
      wrapper = mount(<ViewProfile />);
    });
  });
  it('Should render with no errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should render a link', () => {
    expect(wrapper.text().includes('COMPLETE')).toBe(true);
  });
});
