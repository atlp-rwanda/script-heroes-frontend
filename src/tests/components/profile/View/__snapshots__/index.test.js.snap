// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewProfile/> Should render with no errors 1`] = `
<View>
  <Container
    className="background height text-color"
    tag="div"
  >
    <div
      className="background height text-color container"
    >
      <h3
        className="text-center mb-2"
      >
        Profile Information
      </h3>
      <Row
        className=""
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="row"
        >
          <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col"
            >
              <Table
                borderless={true}
                responsiveTag="div"
                tag="table"
              >
                <table
                  className="table table-borderless"
                >
                  <tr>
                    <td
                      className="text-uppercase text-color"
                    >
                      <i
                        className="fas fa-user mr-3 text-color"
                      />
                      FIRST NAME:
                    </td>
                    <td
                      className="text-capitalize"
                    />
                  </tr>
                  <tr>
                    <td
                      className="text-uppercase text-color"
                    >
                      <i
                        className="fas fa-user mr-3 text-color"
                      />
                      last NAME:
                    </td>
                    <td
                      className="text-capitalize"
                    />
                  </tr>
                  <tr>
                    <td
                      className="text-uppercase text-color"
                    >
                      <i
                        className="fas fa-envelope mr-3 text-color"
                      />
                      email:
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td
                      className="text-uppercase text-color"
                    >
                      <i
                        className="fas fa-transgender mr-3 text-color"
                      />
                      gender:
                    </td>
                    <td
                      className="text-capitalize"
                    />
                  </tr>
                  <tr>
                    <td
                      className="text-uppercase text-color"
                    >
                      <i
                        className="fas fa-calendar-day mr-3 text-color"
                      />
                      Date of birth:
                    </td>
                    <td
                      className="text-capitalize"
                    >
                      Invalid Date
                    </td>
                  </tr>
                  <tr>
                    <td
                      className="text-uppercase text-color"
                    >
                      <i
                        className="fas fa-phone-alt mr-3 text-color"
                      />
                      Phone number:
                    </td>
                    <td
                      className="text-capitalize"
                    />
                  </tr>
                </table>
              </Table>
            </div>
          </Col>
          <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col"
            >
              <Table
                borderless={true}
                responsiveTag="div"
                tag="table"
              >
                <table
                  className="table table-borderless"
                >
                  <tr>
                    <td
                      className="text-uppercase text-color"
                    >
                      <i
                        className="fas fa-flag mr-3 text-color"
                      />
                      country of residence:
                    </td>
                    <td
                      className="text-capitalize"
                    />
                  </tr>
                  <tr>
                    <td
                      className="text-uppercase text-color"
                    >
                      <i
                        className="fas fa-coins mr-3 text-color"
                      />
                      currency:
                    </td>
                    <td
                      className="text-capitalize"
                    />
                  </tr>
                  <tr>
                    <td
                      className="text-uppercase text-color"
                    >
                      <i
                        className="fas fa-language mr-3 text-color"
                      />
                      language:
                    </td>
                    <td
                      className="text-capitalize"
                    />
                  </tr>
                  <tr>
                    <td
                      className="text-uppercase text-color"
                    >
                      <i
                        className="fas fa-building mr-3 text-color"
                      />
                      department:
                    </td>
                    <td
                      className="text-capitalize"
                    />
                  </tr>
                  <tr>
                    <td
                      className="text-uppercase text-color"
                    >
                      <i
                        className="fas fa-user-edit mr-3 text-color"
                      />
                      line manager:
                    </td>
                    <td
                      className="text-capitalize"
                    />
                  </tr>
                </table>
              </Table>
            </div>
          </Col>
        </div>
      </Row>
      <a
        className="float-right text-color none m-1"
        href="/profile/complete/undefined"
      >
        <strong>
           COMPLETE 
        </strong>
      </a>
    </div>
  </Container>
</View>
`;
